### ISSUE TITLE
A few improvements to the tools/modules/update.sh script: [phab:D1171]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1540]
_Created by **pcc** at 2016-12-03 21:16:32_: 

- Don't clone clang-tools-extra, kythe doesn't use it unless I'm mistaken
- Use the github mirrors instead of llvm.org, as they are generally faster
- Use ninja if available to build LLVM

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-12-05 20:37:45_:

I'm dubious about the assertion that "anyone who cares about build speed should be using ninja anyway". I also don't have ninja installed on my various machines and am hesistant to add another (albeit optional) dependency. Is there a reason why `make` should be pessimized in this way?

**Comment by kythe** [comment]
_**pcc** commented inline (context lost during migration) at 2016-12-05 20:55:58_:

That seems like a fair position. Would you be happy with ninja as an optional dependency as long as it doesn't regress make?

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-12-05 21:02:07_:

Yes, that's fine (and the rest of your changes LGTM and are appreciated).

**Comment by kythe** [comment]
_**pcc** wrote at 2016-12-05 21:57:01_:

  - Restore the logic for guessing number of cores
  - Document the optional ninja dependency

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-12-05 22:57:21_:

`build.*.*.*` should probably still be an option; otherwise once we need to do a version bump, people will be left with old build directories.

**Comment by kythe** [comment]
_**zarko** wrote at 2016-12-05 22:57:21_:

Tested this in existing and no-prebuild configurations on Linux and OS X (and no-prebuild with Ninja on Linux).

This is causing a build failure on OS X:

```
cat .../kythe/cxx/extractor/testdata/metadata/test.log
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
-----------------------------------------------------------------------------
sed: 1: "/-target/,+1d": expected context address
```

**Comment by kythe** [comment]
_**pcc** commented inline (context lost during migration) at 2016-12-05 23:04:02_:

But `build.*.*` matches anything that `build.*.*.*` would match, right?

**Comment by kythe** [comment]
_**pcc** wrote at 2016-12-05 23:04:02_:

>>! In (earlier comment), zarko wrote:
> Tested this in existing and no-prebuild configurations on Linux and OS X (and no-prebuild with Ninja on Linux).
> 
> This is causing a build failure on OS X:
> 
> ```
> cat .../kythe/cxx/extractor/testdata/metadata/test.log
> exec ${PAGER:-/usr/bin/less} "$0" || exit 1
> -----------------------------------------------------------------------------
> sed: 1: "/-target/,+1d": expected context address
> ```

I'll see if I can reproduce this on OS X. But are you sure that my change was the culprit here? I don't see what that test has to do with my change.

**Comment by kythe** [comment]
_**zarko** wrote at 2016-12-05 23:06:18_:

#1536 fixed the sed issue; your diff was cut ~2 minutes before it was landed.

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2016-12-05 23:08:12_:

As it turns out, this is not some magic `*` that only matches a field, but the normal sort of find `*`. Never mind.

**Comment by kythe** [comment]
_**zarko** closed via commit fe8961e78df4fe3b530069394cd97fee42bb8493 at 2016-12-06 00:02:11_
